src/lib/plugin_apis/crypto.api

```diff
@@ -1111,6 +1112,193 @@ gboolean bd_crypto_luks_set_uuid (const gchar *device, const gchar *uuid, GError
+ * @size    size of the device being reencrypted
+ * @offset  current offset
+ *
+ * A callback function called during reencryption to report progress. Also used to possibly stop reencryption.
+ *
+ * Returns: 0, if the reencryption should continue.
+ *          A non-zero value to stop the reencryption
+ */
+typedef int (*BDCryptoLUKSReencryptProgFunc) (guint64 size, guint64 offset);
+
+typedef enum {
+    BD_CRYPTO_LUKS_REENCRYPT_NONE = 0,
+    BD_CRYPTO_LUKS_REENCRYPT_CLEAN,
+    BD_CRYPTO_LUKS_REENCRYPT_CRASH,
+    BD_CRYPTO_LUKS_REENCRYPT_INVALID
+} BDCryptoLUKSReencryptStatus;
+
+typedef enum {
+    BD_CRYPTO_LUKS_REENCRYPT = 0,
+    BD_CRYPTO_LUKS_ENCRYPT,
+    BD_CRYPTO_LUKS_DECRYPT,
+} BDCryptoLUKSReencryptMode;
+
+/**
+ * bd_crypto_luks_reencrypt:
+ * @device:     device to reencrypt. Either an active device name for online reencryption, or a block device for offline reencryption.
+ *              Must match the @params's "offline" parameter
+ * @params:     reencryption parameters
+ * @context:    key slot context to unlock @device. The newly created keyslot will use the same context
+ * @prog_func: (scope call) (nullable): progress function. Also used to possibly stop reencryption
+ * @error: (out) (optional): place to store error (if any)
+ *
+ * Reencrypts @device. This could mean a change of cipher, cipher mode, or volume key, based on @params
+ *
+ * Returns: true,  if the reencryption was successful or gracefully stopped with @prog_func.
+ *          false, if an error occurred.
+ *
+ * Supported @context types for this function: passphrase
+ *
+ * Tech category: %BD_CRYPTO_TECH_LUKS-%BD_CRYPTO_TECH_MODE_MODIFY
+ */
+gboolean bd_crypto_luks_reencrypt(const gchar *device, BDCryptoLUKSReencryptParams *params, BDCryptoKeyslotContext *context, BDCryptoLUKSReencryptProgFunc prog_func, GError **error);
+
+/**
+ * bd_crypto_luks_reencrypt_status:
+ * @device:                     an active device name or a block device
+ * @mode: (out):                the exact operation in the "reencryption family"
+ *                                Has no meaning if the return value is BD_CRYPTO_LUKS_REENCRYPT_NONE
+ * @error: (out) (optional):    place to store error (if any)
+ *
+ * Returns: state of @device's reencryption
+ *
+ * Tech category: %BD_CRYPTO_TECH_LUKS-%BD_CRYPTO_TECH_MODE_QUERY
+ */
+BDCryptoLUKSReencryptStatus bd_crypto_luks_reencrypt_status (const gchar *device, BDCryptoLUKSReencryptMode *mode, GError **error);
+
+/**
+ * bd_crypto_luks_reencrypt_resume:
+ * @device:     device with a stopped reencryption. An active device name or a block device
+ * @context:    key slot context to unlock @device
+ * @prog_func: (scope call) (nullable): progress function. Also used to possibly stop reencryption
+ * @error: (out) (optional):    place to store error (if any)
+ *
+ * Returns: true,  if the reencryption finished successfully or was gracefully stopped with @prog_func.
+ *          false, if an error occurred.
```

Comment:
Just like above, would be good to know if this call blocks until the operation is finished or if it just resumes an operation in the background.